### DISCUSSION TITLE
Confused about embedding models w/ Together AI.

### DISCUSSION THREAD

**Discussion by mweel1** [tryAGI/LangChain#406] [General]
I'm confused how to use Together AI and what to setup for its embedding model.

<img width="761" alt="image" src="https://github.com/user-attachments/assets/0233d638-2794-45da-b1ba-4d90155a7a44">

Do I have to use OpenAI to do my embeddings and then Together to run the prompt?

Thanks in advance

**Comment by HavenDV** [comment]
https://docs.together.ai/docs/embedding-models
Yes, we don't have Enum implemented for Together embedded models at the moment. But you can still do it via `OpenAiEmbeddingModel(provider, id)` by specifying the appropriate Id like `"togethercomputer/m2-bert-80M-8k-retrieval"` and using `TogetherProvider` as the first argument.

**Reply by HavenDV** [reply]
Most implementations of providers like Together and Anyscale use the OpenAI SDK internally, just with a custom Base URL and other model ids.
